mountpoint-s3/src/prefetch/caching_stream.rs

```diff
@@ -46,6 +48,9 @@ where
         if_match: ETag,
         range: RequestRange,
         _preferred_part_size: usize,
+        _in_memory_cache: InMemoryCache,
+        _parsed_metadata: MetadataRef,
+        _raw_metadata: RawMetadataRef,
```

Comment:
We don't use these medata at all in caching stream and client stream which make me think we should initialise them in `parquet_stream.rs` rather than `prefetch.rs`. I remember you also have a plan to save it somewhere to be reused by multiple file handles and that place is not going to be`prefetch.rs` anyway, right?

Reply:
My idea is that the cache will probably be held in the "prefetch.rs" level (for multiple file handles), which is why I passed it in as parameters, I'm thinking of consolidating the parameters though. I'll think more on how we can avoid this and get back on this.